Integration/AndroidAAR/BabylonNative/src/main/java/BabylonNative/BabylonView.java

```diff
@@ -0,0 +1,113 @@
+        super(context);
+        init(viewDelegate);
+        BabylonNative.Wrapper.initEngine(context.getResources().getAssets());
+    }
+
+    private void init(ViewDelegate viewDelegate) {
+        SurfaceHolder holder = getHolder();
+        holder.addCallback(this);
+        setOnTouchListener(this);
+        mViewDelegate = viewDelegate;
+    }
+
+    public void loadScript(String path) {
+        BabylonNative.Wrapper.loadScript(path);
+    }
+
+    public void eval(String source, String sourceURL) {
+        BabylonNative.Wrapper.eval(source, sourceURL);
+    }
+
+    public void onPause() {
+        setVisibility(View.GONE);
+        BabylonNative.Wrapper.activityOnPause();
+    }
+
+    public void onResume() {
+        BabylonNative.Wrapper.activityOnResume();
+    }
+
+    /**
+     * This method is part of the SurfaceHolder.Callback interface, and is
+     * not normally called or subclassed by clients of BabylonView.
+     */
+    public void surfaceCreated(SurfaceHolder holder) {
+        BabylonNative.Wrapper.surfaceCreated(getHolder().getSurface());
+        if (!this.mViewReady) {
+            mViewDelegate.onViewReady();
+            mViewReady = true;
+        }
+    }
+
+    /**
+     * This method is part of the SurfaceHolder.Callback interface, and is
+     * not normally called or subclassed by clients of BabylonView.
+     */
+    public void surfaceDestroyed(SurfaceHolder holder) {
+    }
+
+    /**
+     * This method is part of the SurfaceHolder.Callback interface, and is
+     * not normally called or subclassed by clients of BabylonView.
+     */
+    public void surfaceChanged(SurfaceHolder holder, int format, int w, int h) {
+        BabylonNative.Wrapper.surfaceChanged(w, h, getHolder().getSurface());
+    }
+
+    public interface ViewDelegate {
+        void onViewReady();
+    }
+
+    @Override
+    public boolean onTouch(View v, MotionEvent event)
+    {
+        float mX = event.getX();
+        float mY = event.getY();
+        switch (event.getAction()) {
+            case MotionEvent.ACTION_DOWN:
+                BabylonNative.Wrapper.setTouchInfo(mX, mY, true);
+                break;
+            case MotionEvent.ACTION_MOVE:
+                BabylonNative.Wrapper.setTouchInfo(mX, mY, true);
+                break;
+            case MotionEvent.ACTION_UP:
+                BabylonNative.Wrapper.setTouchInfo(mX, mY, false);
+                break;
+        }
+        return true;
+    }
+
+    @Override
+    protected void finalize() throws Throwable {
+        BabylonNative.Wrapper.finishEngine();
+    }
+
+    /**
+     * This method is part of the SurfaceHolder.Callback2 interface, and is
+     * not normally called or subclassed by clients of BabylonView.
+     */
+    @Deprecated
+    @Override
+    public void surfaceRedrawNeeded(SurfaceHolder holder) {
+        // Since we are part of the framework we know only surfaceRedrawNeededAsync
+        // will be called.
```

Comment:
I don't really understand this comment. The default behavior of surfaceRedrawNeededAsync is to just call surfaceRedrawNeeded, so I guess that means we are saying nothing happens on surfaceRedrawNeeded/surfaceRedrawNeededAsync?

Reply:
Comment updated